### PR TITLE
[Enhancement] Update date column cast

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ImplicitCastRule.java [hunk 1]
```diff
@@ -113,12 +113,18 @@ public ScalarOperator visitBinaryPredicate(BinaryPredicateOperator predicate,
             if (op.isPresent()) {
                 predicate.getChildren().set(0, op.get());
                 return predicate;
+            } else if (rightChild.getType().isDateType()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ImplicitCastRule.java:116]: Comment why date type is special.

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/ImplicitCastRule.java|4|6|66.67%|[118, 119]|

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
@mergeify backport branch-2.1 branch-2.2 branch-2.3

**Comment by Seaven** [comment]
@mergefiy backport branch-2.1

**Comment by Seaven** [comment]
@Mergeify backport branch-2.1

**Pull request merged**